omniqueue/src/backends/gcp_pubsub.rs

```diff
@@ -255,7 +255,9 @@ async fn subscription(client: &Client, subscription_id: &str) -> Result<Subscrip
     Ok(subscription)
 }
 
-impl_queue_consumer!(GcpPubSubConsumer, Payload);
+impl_queue_consumer!(for GcpPubSubConsumer {
+    type Payload = Payload;
+});
```

Comment:
It's surprising to me there's no upper bound for GCP. I wonder if it's just not documented, but there's a de facto limit. It could also be that they clamp it on their end. 🤷🏻

Reply:
Well, their docs say

> The Pub/Sub system may return fewer than the `maxMessages` requested even if there are more messages available in the backlog.

so I think they just do the sensible thing and clamp server-side to whatever makes sense to them ^^